pkg/operator/ceph/object/user/controller.go

```diff
@@ -269,27 +269,28 @@ func (r *ReconcileObjectStoreUser) reconcile(request reconcile.Request) (reconci
 }
 
 func (r *ReconcileObjectStoreUser) reconcileCephUser(cephObjectStoreUser *cephv1.CephObjectStoreUser) (reconcile.Result, error) {
-	err := r.createCephUser(cephObjectStoreUser)
+	err := r.createorUpdateCephUser(cephObjectStoreUser)
 	if err != nil {
-		return reconcile.Result{}, errors.Wrapf(err, "failed to create object store user %q", cephObjectStoreUser.Name)
+		return reconcile.Result{}, errors.Wrapf(err, "failed to create/update object store user %q", cephObjectStoreUser.Name)
 	}
 
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileObjectStoreUser) createCephUser(u *cephv1.CephObjectStoreUser) error {
+func (r *ReconcileObjectStoreUser) createorUpdateCephUser(u *cephv1.CephObjectStoreUser) error {
 	logger.Infof("creating ceph object user %q in namespace %q", u.Name, u.Namespace)
 	user, rgwerr, err := object.CreateUser(r.objContext, r.userConfig)
 	if err != nil {
 		if rgwerr == object.ErrorCodeFileExists {
-			objectUser, _, err := object.GetUser(r.objContext, r.userConfig.UserID)
+			objectUser, _, err := object.UpdateUser(r.objContext, r.userConfig)
 			if err != nil {
 				return errors.Wrapf(err, "failed to get details from ceph object user %q", objectUser.UserID)
 			}
 
 			// Set access and secret key
 			r.userConfig.AccessKey = objectUser.AccessKey
 			r.userConfig.SecretKey = objectUser.SecretKey
+			logger.Infof("ceph object user %q updated with display name %s", u.Name, *objectUser.DisplayName)
```

Comment:
It won't be updated every time, right? Either we should print at info level only when it changes, or write to debug
```suggestion
			logger.Debugf("ceph object user %q updated with display name %s", u.Name, *objectUser.DisplayName)
```

Reply:
Done